src/include/cursor_inline.h

```diff
@@ -440,6 +440,39 @@ __wt_cursor_func_init(WT_CURSOR_BTREE *cbt, bool reenter)
     return (0);
 }
 
+/*
+ * __wt_cursor_free_cached_memory --
+ *     If a cached cursor is still holding memory, free it now.
+ */
+static WT_INLINE void
+__wt_cursor_free_cached_memory(WT_CURSOR *cursor)
+{
+    WT_SESSION_IMPL *session;
+
+    if (F_ISSET(cursor, WT_CURSTD_CACHED_WITH_MEM)) {
+        session = CUR2S(cursor);
+
+        /* Don't keep buffers allocated for cached cursors. */
+        __wt_buf_free(session, &cursor->key);
+        __wt_buf_free(session, &cursor->value);
+
+        /* Discard the underlying WT_CURSOR_BTREE buffers. */
+        __wt_btcur_cache((WT_CURSOR_BTREE *)cursor);
```

Comment:
Does it make sense to have this in an additional function call? It's freeing extra buffers which we're also doing here.

Side not: This isn't related to your PR but the name of the function `__wt_btcur_cache` seems off.

Reply:
__wt_btcur_cache operates on with things in the btree cursor, so the function makes sense to be in the btree module.  I think I'll rename it __wt_btcur_free_cached_memory.

Reply:
Much better name, thank you!